File: core/test/org.ebayopensource.vjet.test.core.jstojava/src/org/ebayopensource/dsf/jst/validation/vjo/typecheck/array/ArrayType1Tester.js
/* @com.ebay.dsf.resource.utils.CodeGen("VjoGenerator") */

vjo.ctype('org.ebayopensource.dsf.jst.validation.vjo.typecheck.array.ArrayType1Tester') //< public
.props({
    //> public void a()
    a:function(){
    	var pros = ["SHSH", 2, 23.42, true]; //< Array
    },
    
    //> public void a1()
    a1:function(){
    	var pros = ["1", 2];
    },
    
 	//> public void a2()
    a2:function(){
    	var pros = ["1", '2']; //<String[]
    },
    
    //> public void a3()
    a3:function(){
    	var pros = ["1", '2', true];
    },
    
    //> public void a5()
    a5:function(){
    	var pros = ["1", 23.02 ];
    }
})
.endType();

File: core/test/org.ebayopensource.vjet.test.core.jstojava/src/org/ebayopensource/dsf/jst/validation/vjo/typecheck/array/ArrayType1Tester.java
/*******************************************************************************
 * Copyright (c) 2005-2011 eBay Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/
/* 
 * $Id: ArrayType1Tester.java, May 7, 2009, 6:34:28 PM, liama. Exp$
 *
 * Copyright (c) 2006-2009 Ebay Technologies. All Rights Reserved.
 * This software program and documentation are copyrighted by Ebay 
 * Technologies.
 */
package org.ebayopensource.dsf.jst.validation.vjo.typecheck.array;




import java.util.List;

import org.ebayopensource.dsf.jsgen.shared.validation.vjo.VjoSemanticProblem;
import org.ebayopensource.dsf.jst.validation.vjo.VjoValidationBaseTester;
import org.junit.Before;
import org.junit.Test;




/**
 * For test the in array types in Equality Operator.
 * 
 * a:function(){ var pros = ["1", 2, 23.42, true]; },
 * 
 * //> public void a1() a1:function(){ var pros = ["1", 2]; },
 * 
 * //> public void a2() a2:function(){ var pros = ["1", '2']; },
 * 
 * //> public void a3() a3:function(){ var pros = ["1", '2', true]; },
 * 
 * //> public void a5() a5:function(){ var pros = ["1", 23.02 ]; }
 * 
 * @author <a href="mailto:liama@ebay.com">liama</a>
 * @since JDK 1.5
 */
//@ModuleInfo(value="DsfPrebuild",subModuleId="JsToJava")
//@Category( { P3, FAST, UNIT })
public class ArrayType1Tester extends VjoValidationBaseTester {

    @Before
    public void setUp() {
        expectProblems.clear();
    }

    @Test
    //@Category( { P3, FAST, UNIT })
    //@Description("Test assign value to array children's type mismatch")
    public void testCompartibleType1() {
        List<VjoSemanticProblem> problems = getVjoSemanticProblem(
                "ArrayType1Tester.js", this.getClass());
        assertProblemEquals(expectProblems, problems);
    }
}
